### PR TITLE
Document Quicksilver Secret Storage

### LINKED ISSUE #5483: Automating and Integrating your Pantheon WebOps Workflow with Quicksilver Platform Hooks Doc Update
Re: [Automating and Integrating your Pantheon WebOps Workflow with Quicksilver Platform Hooks](https://pantheon.io/docs/quicksilver)

Priority: Low

## Issue Description:
This page about Quicksilver doesn't even have `secrets.json` on the page, even though it's an essential part of using Quicksilver to integrate with third-party APIs. We frequently refer to it in training and give specific examples in the [Quicksilver Examples Repo](https://github.com/pantheon-systems/quicksilver-examples). We also refer to it within integration guides on Docs (like [Integrate Jira](https://pantheon.io/docs/guides/jira) and [Integrate Trello](https://pantheon.io/docs/guides/trello)). This feels important enough to provide some documentation around.

## Suggested Resolution
At a minimum, we should add a section to the Quicksilver doc to define secrets.json and how it is used. I could see this having enough meat to get its own page, but I'll leave that to your expertise to decide.

### PR DESCRIPTION
**Pull request by alexfornuto**
Closes #5483

## Effect
PR includes the following changes:
- New section in quicksilver.md covering storage of secret data.

## Remaining Work
- [x] Technical review (@davidneedham, @greg-1-anderson)
- [x] Copy Review

## Post Launch
**Do not remove** - To be completed by the docs team upon merge:
- [ ] Update Status Report
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5573 | Difference |
| --- | --- | --- | --- |
| Performance Score | `59` | `41` | `-18` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5573` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18982/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18982/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

File: source/content/quicksilver.md [hunk 1]
```diff
@@ -58,6 +57,13 @@ You can hook into the following workflows:
 | `sync_code`                            | Push code via Git or commit OSD/SFTP changes via Pantheon Dashboard | Dev or Multidev            |                                             |
 | `create_cloud_development_environment` | Create Multidev environment                                         | Multidev                   | `after` stage valid, `before` stage invalid |
 
+## Secrets
+
+Your script may require tokens, passwords, or other information that should be protected. These values should be stored in securely. Some methods for secure storage include:
+
+- A key management service like [Lockr](/guides/lockr),
+- A storage solution in your site's [private files path](/private-paths#private-path-for-files),
+- A secrets file managed by the [Terminus Secrets Plugin](https://github.com/pantheon-systems/terminus-secrets-plugin).
```
Comment (greg-1-anderson) [inline source/content/quicksilver.md]: The second and third bullets are pretty much the same -- or, more to the point, the third bullet is just one example of a way to achieve the goals of the second bulltet.

Maybe remove the third bullet, and instead add a bit of text after the list:

> The Terminus Secrets Plugin provides a convenient way to manage secret data in json files in your site's private files path.

**Comment by greg-1-anderson** [comment]
Looks good. It might also be worth noting somewhere that if the dashboard function to copy the files directory from one environment to another is used, that will overwrite secrets in the destination environment.

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5573 | Difference |
| --- | --- | --- | --- |
| Performance Score | `60` | `53` | `-7` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5573` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18990/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/18990/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

**Comment by alexfornuto** [comment]
> I was expecting to an example implementation but realize from our conversation earlier than there may not be a clear "standard".

There is no standard, other than the standard of JSON formatting.

> Regardless, we have examples that exist in other projects. Is it worthwhile to link to one or two of those here as a "see it in action" cta?

Each example used in another page is a version of the example in the [plugin README](https://github.com/pantheon-systems/terminus-secrets-plugin#usage), only specific to the service covered in that page.

I think any more detail or example on the `secrets.json` file should go, as you suggested in the original issue, into a new doc on the Terminus Secrets Plugin, as `secrets.json` is a feature of that tool and not Quicksilver itself.

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5573 | Difference |
| --- | --- | --- | --- |
| Performance Score | `59` | `44` | `-15` |
| Best Practice Score | `86` | `86` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-5573` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/19148/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/19148/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | Live | pr-5573 | Difference |
| --- | --- | --- | --- |
| Performance Score | `52` | `0` | `-52` |
| Best Practice Score | `86` | `0` | `-86` |
| SEO Score | `96` | `0` | `-96` |
| Accessibility Score | `96` | `0` | `-96` |

Lighthouse Audit Reports:

- [`pr-5573` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/19182/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/19182/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)

**Pull request merged**